Log FocusOut events

psychon opened the pull request:
Currently, i3 only logs FocusIn events. Thus, a debug log tells us when
some window gets the focus. However, we don't know when it loses the
focus. This commit remedies this by adding some log messages for this.

To decide what to log, I just copied some code from handle_focus_in().
No idea how relevant all these details are.

Signed-off-by: Uli Schlachter <psychon@znc.in>
Idea-in-context-of: https://github.com/i3/i3/issues/4532

orestisfl reviewed (changes requested):
I would prefer if this function produced just one log line. For FocusIn it makes sense that there are mutliple because it helps follow the flow of the code in the logs.

My suggestion is to get all the information you need in const variables and make a single DLOG call in the end.

psychon committed:
Log FocusOut events

Currently, i3 only logs FocusIn events. Thus, a debug log tells us when
some window gets the focus. However, we don't know when it loses the
focus. This commit remedies this by adding some log messages for this.

Since I had no idea what to log, this just logs all the fields from the
event plus tries to find a name for the window.

Signed-off-by: Uli Schlachter <psychon@znc.in>
Idea-in-context-of: https://github.com/i3/i3/issues/4532

psychon commented:
Updated. This now just logs all the fields of the event.

orestisfl reviewed (approved):
Thanks!
